<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GeneralizedSpectrumGUIClass</class>
 <widget class="QMainWindow" name="GeneralizedSpectrumGUIClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1280</width>
    <height>960</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>GeneralizedSpectrumGUI</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QCustomPlot" name="CustomPlot" native="true">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>1051</width>
      <height>311</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>1071</x>
      <y>12</y>
      <width>210</width>
      <height>941</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="menuLayout">
     <property name="spacing">
      <number>5</number>
     </property>
     <property name="sizeConstraint">
      <enum>QLayout::SetDefaultConstraint</enum>
     </property>
     <item>
      <widget class="QPushButton" name="openImageButton">
       <property name="minimumSize">
        <size>
         <width>150</width>
         <height>34</height>
        </size>
       </property>
       <property name="text">
        <string>Обработать картинку</string>
       </property>
       <property name="flat">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="saveImageButton">
       <property name="minimumSize">
        <size>
         <width>150</width>
         <height>34</height>
        </size>
       </property>
       <property name="text">
        <string>Сохранить картинку</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="refreshSkeletonButton">
       <property name="minimumSize">
        <size>
         <width>150</width>
         <height>34</height>
        </size>
       </property>
       <property name="text">
        <string>Обновить скелет</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="savePolyButton">
       <property name="text">
        <string>Сохранить полигоны</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="labePruning">
       <property name="text">
        <string>Параметр стрижки:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="pruning">
       <property name="maximum">
        <number>100000000</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="labelArea">
       <property name="text">
        <string>Игнорировать площадь меньше:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="area">
       <property name="maximum">
        <number>100000000</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="labelStep">
       <property name="text">
        <string>Шаг сетки:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QDoubleSpinBox" name="step">
       <property name="minimum">
        <double>0.010000000000000</double>
       </property>
       <property name="maximum">
        <double>100.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.100000000000000</double>
       </property>
       <property name="value">
        <double>0.100000000000000</double>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="labelRadius">
       <property name="text">
        <string>Радиус примитива:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QDoubleSpinBox" name="radius">
       <property name="maximum">
        <double>999.990000000000009</double>
       </property>
       <property name="singleStep">
        <double>0.100000000000000</double>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="lacunasButton">
       <property name="text">
        <string>Лакуны и триангуляция</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="tracingButton">
       <property name="text">
        <string>Инкрементное прослеживание</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <layout class="QVBoxLayout" name="xDegLayout">
         <item>
          <widget class="QLabel" name="xDegLabel">
           <property name="text">
            <string>Степень X:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="xDeg"/>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="yDegLayout">
         <item>
          <widget class="QLabel" name="yDegLabel">
           <property name="text">
            <string>Степень Y:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="yDeg"/>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QCheckBox" name="invertCheckBox">
       <property name="text">
        <string>Инвертировать картинку</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="filenameLabel">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>Имя файла:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="timeLabel">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>Время:</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="verticalSpacer">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QLabel" name="labelScale">
       <property name="text">
        <string>Масштаб:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSlider" name="scale">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimum">
        <number>10</number>
       </property>
       <property name="maximum">
        <number>1000</number>
       </property>
       <property name="singleStep">
        <number>10</number>
       </property>
       <property name="pageStep">
        <number>100</number>
       </property>
       <property name="value">
        <number>100</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="imageCB">
       <property name="text">
        <string>Изображение</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="bonesCB">
       <property name="autoFillBackground">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>Ребра</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="circlesCB">
       <property name="text">
        <string>Круги</string>
       </property>
       <property name="checked">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="contoursCB">
       <property name="text">
        <string>Контуры</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="MyPainter" name="Painter">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>330</y>
      <width>1051</width>
      <height>621</height>
     </rect>
    </property>
   </widget>
   <zorder>layoutWidget</zorder>
   <zorder>CustomPlot</zorder>
   <zorder>Painter</zorder>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header>../PatternSpectrumGUI/qcustomplot.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>MyPainter</class>
   <extends>QGraphicsView</extends>
   <header>../PatternSpectrumGUI/mypainter.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
